torchsde/_core/adjoint_sde.py

```diff
@@ -99,21 +139,27 @@ def f_corrected_diagonal(self, t, y_aug):  # For Ito diagonal.
                 inputs=[y] + self._params,
                 grad_outputs=adj_y,
                 allow_unused=True,
-                create_graph=True
+                retain_graph=True,
+                create_graph=requires_grad
             )
+            if not requires_grad:
```

Comment:
For the sake of consistency, do you think it'd be better to put this detach statement near the bottom just before the `return ...` statement?

Reply:
Done.